routing/control_tower.go

```diff
@@ -191,23 +203,22 @@ func (p *controlTower) SubscribePayment(paymentHash lntypes.Hash) (
 	// a subscriber, because we can send the result on the channel
 	// immediately.
 	case channeldb.StatusSucceeded:
-		event.Success = true
-		event.Preimage = *payment.PaymentPreimage
-		event.Route = &payment.Attempt.Route
+		event = createSuccessResult(
+			&payment.Attempt.Route, *payment.PaymentPreimage,
+		)
 
 	// Payment already failed. It is not necessary to register as a
 	// subscriber, because we can send the result on the channel
 	// immediately.
 	case channeldb.StatusFailed:
-		event.Success = false
-		event.FailureReason = *payment.Failure
+		event = createFailedResult(*payment.Failure)
 
 	default:
 		return false, nil, errors.New("unknown payment status")
 	}
 
 	// Write immediate result to the channel.
-	c <- event
+	c <- *event
```

Comment:
Why not leave the declaration above is a regular type since we immediately deference this pointer?

Reply:
Fixed